be/src/exec/vectorized/aggregate/aggregate_blocking_node.cpp

```diff
@@ -176,8 +176,7 @@ std::vector<std::shared_ptr<pipeline::OperatorFactory> > AggregateBlockingNode::
             // 2. Otherwise, add LocalExchangeOperator
             // to shuffle multi-stream into #degree_of_parallelism# streams each of that pipes into AggregateBlockingSinkOperator.
             bool need_local_shuffle = true;
-            if (operators_with_sink.size() == 1 &&
-                typeid(*operators_with_sink[0]) == typeid(pipeline::ExchangeSourceOperatorFactory)) {
+            if (typeid(*operators_with_sink[0]) == typeid(pipeline::ExchangeSourceOperatorFactory)) {
```

Comment:
suggest to write as
```
if (auto* exchange_op = dynamic_cast<pipeline::ExchangeSourceOperatorFactory*>(operators_with_sink[0].get())) {
}
```

this style is very common and clear

Reply:
Good suggestion!